geo/src/algorithm/stitch.rs

```diff
@@ -0,0 +1,582 @@
+use std::collections::HashMap;
+
+use geo_types::{Coord, Line, LineString, MultiPolygon, Polygon, Triangle};
+
+use crate::{Contains, GeoFloat, LinesIter, Winding};
+
+// ========= Error Type ============
+
+#[derive(Debug)]
+pub enum LineStitchingError {
+    IncompleteRing,
+    InvalidGeometry,
+    MissingParent,
+    NoExtremum,
+}
+
+impl std::fmt::Display for LineStitchingError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+impl std::error::Error for LineStitchingError {}
+
+pub(crate) type PolygonStitchingResult<T> = Result<T, LineStitchingError>;
+
+// ========= Main Algo ============
+
+/// Trait to stitch together split up polygons.
+pub trait Stitch<T: GeoFloat> {
+    /// This stitching only happens along identical edges which are located in two separate
+    /// geometries.
+    ///
+    /// ┌─────x        ┌─────┐
+    /// │    /│        │     │
+    /// │   / │        │     │
+    /// │  /  │  ───►  │     │
+    /// │ /   │        │     │
+    /// │/    │        │     │
+    /// x─────┘        └─────┘
```

Comment:
Should this be wrapped in a codeblock to force monospace text?

Reply:
Will do, thanks 👍🏼